examples/cross-package/ctrl/ctrl.go

```diff
@@ -0,0 +1,38 @@
+package ctrl
+
+import (
+	"context"
+
+	"github.com/sunboyy/repogen/examples/cross_package"
+)
+
+//go:generate repogen -dest=../repo/user_repo.go -dest-pkg=repo -model-dir=../ -model=UserModel -repo=UserRepository
+
+// UserRepository is an interface that describes the specification of querying
+// user data in the database.
+type UserRepository interface {
+	// InsertOne stores userModel into the database and returns inserted ID
+	// if insertion succeeds and returns error if insertion fails.
+	InsertOne(ctx context.Context, userModel *cross_package.UserModel) (interface{}, error)
+	// FindByUsername queries user by username. If a user with specified
+	// username exists, the user will be returned. Otherwise, error will be
+	// returned.
+	FindByUsername(ctx context.Context, username string) (*cross_package.UserModel, error)
+
+	// UpdateDisplayNameByID updates a user with the specified ID with a new
+	// display name. If there is a user matches the query, it will return
+	// true. Error will be returned only when error occurs while accessing
+	// the database.
+	UpdateDisplayNameByID(ctx context.Context, displayName string, id string) (bool, error)
```

Comment:
Try `UpdateByID(ctx context.Context, user *cross_package.UserModel, id string) (bool, error)` instead. This should prove that it still works when the whole model is updated.